soroban-env-common/env.json

```diff
@@ -902,6 +902,18 @@
                         }
                     ],
                     "return": "Void"
+                },
+                {
+                    "export": "8",
+                    "name": "update_current_contract_wasm",
+                    "args": [
+                        {
+                            "name": "hash",
+                            "type": "BytesObject"
+                        }
+                    ],
+                    "return": "Void",
+                    "docs": "Replaces the executable of the current contract with the provided Wasm code identified by a hash. Wasm entry corresponding to the hash has to already be present in the ledger. The update happens only after the current contract invocation has sucessfully finished, so this can be safely called in the middle of a function."
```

Comment:
I don't quite understand the last sentence. What does it mean "The update happens only after the current contract invocation has sucessfully finished"? 
In your test example the host calls the wasm `update` function -> the VM calls host `update_current_contract_wasm` -> the host updates the contract source -> `update` function continues and returns `123`. 
So the update actually happens before the invocation finishes?

Reply:
The update effectively happens only after the invocation of the current function has finished. The code just updates the ledger entry. If the contract fails, the entry will be rolled back. During the execution we still will be running the current contract code. The calls to this contract following the invocation with update will use the new implementation.